### PR TITLE
[MM-34757] Cloud FF 2021-04-12

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

Cloud FF from master to cloud for 2021-04-12

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-34757

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by marianunez — APPROVED** [review]

**Review by jwilander — APPROVED** [review]

File: store/sqlstore/upgrade_test.go [hunk 1]
```diff
@@ -105,140 +103,142 @@ func TestSaveSchemaVersion(t *testing.T) {
...
+//	m.LastPostAt = lastPostAt
+//	m.CreatorId = creatorId
+//	m.DisplayName = "Name"
+//	m.Name = "zz" + model.NewId() + "b"
+//	m.Type = model.CHANNEL_OPEN
+//	return ss.Channel().Save(&m, -1)
+//}
+
 func TestMsgCountRootMigration(t *testing.T) {
 	t.Skip("Waiting for this tests to be fixed: https://mattermost.atlassian.net/browse/MM-34792")
```
Comment (ethervoid) [inline store/sqlstore/upgrade_test.go:142]: @jwilander @marianunez I've detected some problems with these tests and I tried to fix them but it takes more time than I have so given that is going to affect Master as well Eli is taking care of them [here](https://mattermost.atlassian.net/browse/MM-34792). I

File: store/sqlstore/upgrade.go [hunk 2]
```diff
@@ -1242,5 +1242,24 @@ func hasMissingMigrationsVersion535(sqlStore *SqlStore) bool {
 		return true
 	}
 
+	// Check if the collapsable threads queries were applied
+	msgRootCount, _, err := sqlStore.getQueryBuilder().Select(`COUNT(*) as countRoot`).
```
Comment (ethervoid) [inline store/sqlstore/upgrade.go:1246]: @reflog I only need your review in this part

**Review by reflog — APPROVED** [review]

**Review by cpanato — APPROVED** [review]
took some time to go over all files 😅 

/lgtm

**Pull request merged**